src/bindings/cpp/tests/tests.hpp

```diff
@@ -37,8 +39,147 @@ using namespace kdb;
-	SUCCEED () << message;
+	SUCCEED () << message
 
 #define succeed_if_same(x, y, message) ASSERT_EQ (x, y) << message
 
+#define compare_keyset(keySet1, keySet2) ASSERT_TRUE (compareKeySet (keySet1, keySet2))
+
+// -- Functions ----------------------------------------------------------------------------------------------------------------------------
+
+/**
+ * @brief This function adds the string representation of a key to an assertion result.
+ *
+ * @param stream This parameter specifies the assertion result to which this function adds a string representation of `key`.
+ * @param key This parameter stores the key this function converts to an assertion result.
+ *
+ * @returns An assertion result containing the string representation of `key`
+ */
+testing::AssertionResult & operator<< (testing::AssertionResult & stream, kdb::Key & key)
+{
+	stream << key.getName () << ": " << (key.isString () ? key.getString () : (key.getBinarySize () == 0 ? "NULL" : "BINARY"));
+	key.rewindMeta ();
+
+	key.rewindMeta ();
+	while (key.nextMeta ())
+	{
+		stream << ", " << key.currentMeta ().getName () << ": " << key.currentMeta ().getString ();
+	}
+	stream << endl;
+	return stream;
+}
+
+/**
+ * @brief This function adds the string representation of a key set to an assertion result.
+ *
+ * @param stream This parameter specifies the assertion result to which this function adds a string representation of `keys`.
+ * @param key This parameter stores the key set this function converts to an assertion result.
+ *
+ * @returns An assertion result containing the string representation of `keys`
+ */
+testing::AssertionResult & operator<< (testing::AssertionResult & stream, kdb::KeySet & keys)
+{
+	for (auto key : keys)
+	{
+		stream << key;
+	}
+	return stream;
+}
+
+/**
+ * @brief This function checks if the meta key set of two keys are equal.
+ *
+ * @param key1 This parameter stores the first meta key set this function uses for comparison.
+ * @param key2 This parameter stores the second meta key set this function uses for comparison.
+ *
+ * @retval true if all meta keys of `key1` and `key2` are equal
+ * @retval false otherwise
+ */
+bool isMetaDataEqual (kdb::Key & key1, kdb::Key & key2)
+{
+	key1.rewindMeta ();
+	key2.rewindMeta ();
+
+	while (key1.nextMeta ())
+	{
+		key2.nextMeta ();
+		if (!key2.currentMeta ()) return false;
+		if (key1.currentMeta ().getName () != key2.currentMeta ().getName ()) return false;
+		if (key1.currentMeta ().getString () != key2.currentMeta ().getString ()) return false;
+	}
+
+	return key1.nextMeta () == key2.nextMeta ();
+}
+
+/**
+ * @brief This function checks if two keys are equal.
+ *
+ * @param key1 This parameter stores the first key this function compares.
+ * @param key2 This parameter stores the second key this function compares.
+ *
+ * @retval true if all attributes of `key1` and `key2` are equal
+ * @retval false otherwise
+ */
+bool isKeyEqual (kdb::Key & key1, kdb::Key & key2)
```

Comment:
maybe we can have this in the cframework and reuse it?

Reply:
While I am definitely a fan of code reuse, there are two major problems here:

- my laziness, and
- the code below is written in C++ (and not C)

.